changelogs/fragments/10227-pacemaker-cluster-and-resource-enhancement.yml

```
@@ -0,0 +1,7 @@
+deprecated_features:
+  - state `cleanup` in pacemaker_cluster - is deprecated and will be moved to pacemaker_resource module in community.general 12.0.0 (https://github.com/ansible-collections/community.general/pull/10227).
+  - state `None` in pacemaker_cluster - is deprecated and will be a required parameter in community.general 12.0.0
+
+minor_changes:
+  - pacemaker_resource - add `state=cleanup` value for cleaning up cluster resources (https://github.com/ansible-collections/community.general/pull/10227).
+  - pacemaker_cluster - add unit tests and "maintenance" state for managing pacemaker maintenance mode (https://github.com/ansible-collections/community.general/pull/10227).
```

Removing `cleanup` from 12.0.0 is too early IMO, I think it should only be removed in 13.0.0. (Making `state` required in 12.0.0 is OK IMO since there is no use-case for not specifying it right now.)
```suggestion
  - pacemaker_cluster - state ``cleanup`` is deprecated and will be removed from the module in community.general 13.0.0. Please use the ``community.general.pacemaker_resource`` module instead (https://github.com/ansible-collections/community.general/pull/10227).
  - pacemaker_cluster - the parameter ``state`` will become a required parameter in community.general 12.0.0 (https://github.com/ansible-collections/community.general/pull/10227).

minor_changes:
  - pacemaker_resource - add ``state=cleanup`` for cleaning up cluster resources (https://github.com/ansible-collections/community.general/pull/10227).
  - pacemaker_cluster - add ``state=maintenance`` for managing pacemaker maintenance mode (https://github.com/ansible-collections/community.general/pull/10227).
```